### PR TITLE
fix(contract): improve gas limit demo

### PR DESCRIPTION
**Pull request by idea404**
Improve gas limit demonstration, adding comments and test utilities

### CODE REVIEW

File: src/GlobalManager.sol [hunk 1]
```diff
@@ -61,18 +61,19 @@ contract GlobalManager is XApp, Ownable {
...
-     * @param gasLimit The max amount of gas used by the trx on destination
      */
-    function removeStake(address user, uint256 amount, uint64 gasLimit) external xrecv {
+    function removeStake(address user, uint256 amount) external xrecv {
         require(isXCall(), "GlobalManager: only xcall");
         require(isSupportedChain(xmsg.sourceChainId), "GlobalManager: chain not found");
         require(xmsg.sender == contractOn[xmsg.sourceChainId], "GlobalManager: invalid sender");
         require(stakeOn[user][xmsg.sourceChainId] >= amount, "GlobalManager: insufficient stake");
 
+        uint64 gasLimit = 200_000; // max gas to spend in call to delegate unstaking, ensuring success
```
Comment (kevinhalliday) [inline src/GlobalManager.sol]: Can move to const outside of function

File: src/LocalStake.sol [hunk 2]
```diff
@@ -63,37 +63,34 @@ contract LocalStake is XApp, Ownable {
 
         emit Staked(msg.sender, amount);
 
-        xcall(
+        uint256 fee = xcall(
             globalChainId,
             globalManagerContract,
             abi.encodeWithSelector(GlobalManager.addStake.selector, msg.sender, amount)
```
Comment (kevinhalliday) [inline src/LocalStake.sol]: Do we have a gas limit for this xcall?

File: src/LocalStake.sol [hunk 3]
```diff
@@ -63,37 +63,34 @@ contract LocalStake is XApp, Ownable {
...
             globalChainId,
             globalManagerContract,
-            abi.encodeWithSelector(GlobalManager.removeStake.selector, msg.sender, amount, gasLimit)
+            abi.encodeWithSelector(GlobalManager.removeStake.selector, msg.sender, amount)
         );
 
         // Fee for this.xunstake callback
         uint256 callbackFee =
-            feeFor(uint64(block.chainid), abi.encodeWithSelector(this.xunstake.selector, msg.sender, amount), gasLimit);
+            feeFor(uint64(block.chainid), abi.encodeWithSelector(this.xunstake.selector, msg.sender, amount));
```
Comment (kevinhalliday) [inline src/LocalStake.sol]: And these ones?

File: test/LocalStake.t.sol [hunk 4]
```diff
@@ -77,47 +77,79 @@ contract LocalStakeTest is Test {
...
+        vm.deal(user, 3 * feeAmount); // Ensuring sufficient ether for fees
+        localToken.transfer(user, stakeAmount);
+
+        vm.startPrank(user);
+        localToken.approve(address(localStake), stakeAmount);
+        localStake.stake{value: feeAmount}(stakeAmount);
         vm.stopPrank();
+
+        uint256 gasUsed = gasleft(); // start gas measure
+        vm.prank(globalManagerAddress);
```
Comment (kevinhalliday) [inline test/LocalStake.t.sol]: I'd move gasLeft under this line, right before the call

File: test/LocalStake.t.sol [hunk 5]
```diff
@@ -77,47 +77,79 @@ contract LocalStakeTest is Test {
...
+        portal.mockXCall(
+            globalChainId, address(localStake), abi.encodeWithSelector(localStake.xunstake.selector, user, stakeAmount)
+        );
+        gasUsed = gasUsed - gasleft(); // consumes gas above the 21_000 min gas for Portal xcalls gasLimit param
```
Comment (kevinhalliday) [inline test/LocalStake.t.sol]: This comment does not add much IMO

File: test/GlobalManager.t.sol [hunk 6]
```diff
@@ -77,7 +77,7 @@ contract GlobalTest is Test {
         address user = address(0x123);
         uint64 chainId = 1;
         address contractAddress = address(0x123);
-        uint64 gasLimit = 100_000;
+        uint64 gasLimit = 200_000;
```
Comment (kevinhalliday) [inline test/GlobalManager.t.sol]: Can we be more descriptive with this name?

File: src/GlobalManager.sol (hunk 1 above)
Comment (idea404) [reply src/GlobalManager.sol]: good idea I'll do that

File: src/GlobalManager.sol [hunk 7]
```diff
@@ -30,6 +30,11 @@ contract GlobalManager is XApp, Ownable {
      */
     uint256 public totalStake;
 
+    /**
+     * @dev Constant max gas to spend in call to delegate unstaking, ensuring success
+     */
+    uint64 private constant MAX_GAS_LIMIT = 200_000;
```
Comment (kevinhalliday) [inline src/GlobalManager.sol]: I would name this something specific to the call we are applying it to.

Also, and you comment here saying what test cases showed gas to be.

**Pull request merged**